Vapor 4 support + custom cache keys (#20)
nodes-vapor/gatekeeper: README.md

in configure.swift:
```swift
// Register middleware
app.middlewares.use(GatekeeperMiddleware())
```

#### Customizing config

> By default `GatekeeperMiddleware` uses `app.gatekeeper.config` as it's configuration.

Reviewer: ```suggestion
By default `GatekeeperMiddleware` uses `app.gatekeeper.config` as its configuration.
```

`GatekeeperHostnameKeyMaker` is used by default.

You can configure which key maker Gatekeeper should use in `configure.swift`:
```swift
app.gatekeeper.keyMakers.use(.hostname) // default
```

### Custom key maker
This is an example of a key maker that uses the user's ID to identify them.
```swift
struct UserIDKeyMaker: GatekeeperKeyMaker {
    public func make(for req: Request) -> EventLoopFuture<String> {
        let userID = try req.auth.require(User.self).requireID()        
        return req.eventLoop.future("gatekeeper_" + userID.uuidString)
    }
}
```

```swift
extension Application.Gatekeeper.KeyMakers.Provider {
    public static var userID: Self {
        .init { app in
            app.gatekeeper.keyMakers.use { _ in UserIDKeyMaker() }
        }
    }
}
```
**configure.swift:**
```swift
app.gatekeeper.keyMakers.use(.userID)
```

## Cache 🗄
Gatekeeper uses the same cache as configured by `app.caches.use()` from Vapor, by default.

> Therefore it is **important** to setup Vapor's cache if you're using this default behaviour. You can use an in-memory cache for Vapor like so:

Reviewer: ```suggestion
Therefore it is **important** to set up Vapor's cache if you're using this default behaviour. You can use an in-memory cache for Vapor like so:
```

**configure.swift**:
```swift
app.cache.use(.memory)
```

### Custom cache
You can override which cache to use by creating your own type that conforms to the `Cache` protocol from Vapor. Use `app.gatekeeper.caches.use()` to configure which cache to use.

## Credits 🏆

> This package is developed and maintained by the Vapor team at [Nodes](https://www.nodesagency.com).

Reviewer: It feels wrong for you not to get any credit here ☺️ 

```suggestion
This package is developed and maintained by the Vapor team at [Nodes](https://www.nodesagency.com). Special thanks goes to @madsodgaard for his work on the Vapor 4 version!
```
